test/freenet/client/filter/OggFilterTest.java

```diff
@@ -24,49 +22,60 @@ public void setUp() {
 
 	@Test
 	public void testEmptyOutputRaisesException() throws IOException {
-		DataInputStream input = new DataInputStream(getClass().getResourceAsStream("./ogg/invalid_header.ogg"));
-		ByteArrayOutputStream output = new ByteArrayOutputStream();
-		try {
-			filter.readFilter(input, output, null, null, null, null);
-			fail("Expected Exception not caught. Output size: "+output.toByteArray().length);
-		} catch(DataFilterException e) {}
+		testResourceFile("./ogg/invalid_header.ogg", (input) -> {
+			assertThrows(
+				DataFilterException.class,
+				() -> filter.readFilter(input, new ByteArrayOutputStream(), null, null, null, null)
+			);
+		});
 	}
 
 	@Test
-	public void testValidSubPageStripped() throws IOException, DataFilterException {
-		DataInputStream input = new DataInputStream(getClass().getResourceAsStream("./ogg/contains_subpages.ogg"));
-		ByteArrayOutputStream output = new ByteArrayOutputStream();
-		try {
-			filter.readFilter(input, output, null, null, null, null);
-		} catch(DataFilterException e) {}
-		assertTrue(Arrays.equals(new byte[]{}, output.toByteArray()));
-		input.close();
-		output.close();
+	public void testValidSubPageStripped() throws IOException {
+		testResourceFile("./ogg/contains_subpages.ogg", (input) -> {
+			try (ByteArrayOutputStream output = new ByteArrayOutputStream()) {
+				assertThrows(
```

Comment:
Hmm, I haven’t run the test but I don’t think the exception is the point of this test, otherwise there’d be a `fail()` after the `readFilter()`.

Reply:
Current impl is more strict and expects an exception to be thrown.
Previous solution with empty `try-catch` may be restored easily, 
but that is not able to capture a moment, when `readFilter()` stops throwing exception for configured input.
There is a special branch in the `readFilter()` method, which reports if no bytes were written,
and `assertThrows` tests that branch as well.
A little bit overlaps with `testEmptyOutputRaisesException()`,
but, in my opinion, more checks are better and those help catch more regressions.

Feel free to reject this change if you do not like it.

Reply:
My point is that a test should be for one specific thing. If throwing the exception is an expected outcome for `contains_subpages.ogg` then that should be its own test, and then you’d have a second test for the stripped subpages which does not expect the exception.

Then again, why is an exception thrown but valid stuff is still expected to be written to the output? That seems like a horrible design choice…